pkg/assetvalidator/validateAssets.go

```diff
@@ -34,34 +54,40 @@ func (v *NexposeAssetValidator) ValidateAssets(ctx context.Context, assets []dom
+// that matches the ScanID of the scan completion event that triggered the pipeline, and also returns the type of scan.
+func (v *NexposeAssetValidator) getScanTimeAndScanType(asset domain.Asset, scanInfo domain.ScanInfo) (time.Time, ScanType, error) {
 	for _, evt := range asset.History {
 		scanTime, err := time.Parse(time.RFC3339, evt.Date)
-		switch evt.Type {
-		case "SCAN":
-			if strconv.FormatInt(evt.ScanID, 10) == scanID {
+		if evt.Type == "SCAN" && (scanInfo.ScanType == string(automated) || scanInfo.ScanType == string(manual) || scanInfo.ScanType == string(scheduled)) {
+			if strconv.FormatInt(evt.ScanID, 10) == scanInfo.ScanID {
 				if err != nil {
-					return time.Time{}, &domain.InvalidScanTime{ScanID: scanID, ScanTime: scanTime, AssetID: asset.ID, AssetIP: asset.IP, AssetHostname: asset.HostName, Inner: err}
+					return time.Time{}, remote, &domain.InvalidScanTime{ScanID: scanInfo.ScanID, ScanTime: scanTime, AssetID: asset.ID, AssetIP: asset.IP, AssetHostname: asset.HostName, Inner: err}
 				}
 				if scanTime.IsZero() {
-					return time.Time{}, &domain.InvalidScanTime{ScanID: scanID, ScanTime: scanTime, AssetID: asset.ID, AssetIP: asset.IP, AssetHostname: asset.HostName, Inner: errors.New("scan time is zero")}
+					return time.Time{}, remote, &domain.InvalidScanTime{ScanID: scanInfo.ScanID, ScanTime: scanTime, AssetID: asset.ID, AssetIP: asset.IP, AssetHostname: asset.HostName, Inner: errors.New("scan time is zero")}
 				}
-				return scanTime, nil
+				return scanTime, remote, nil
+			}
+		}
+		if evt.Type == "AGENT-IMPORT" && scanInfo.ScanType == string(agent) {
+			startScanTime, startErr := time.Parse(time.RFC3339, scanInfo.StartTime)
+			endScanTime, endErr := time.Parse(time.RFC3339, scanInfo.EndTime)
+			if err != nil || startErr != nil || endErr != nil {
+				return time.Time{}, remote, &domain.InvalidScanTime{ScanID: scanInfo.ScanID, ScanTime: scanTime, AssetID: asset.ID, AssetIP: asset.IP, AssetHostname: asset.HostName, Inner: err}
 			}
-		case "AGENT-IMPORT":
-			if time.Since(scanTime).Hours() < 24 {
-				return scanTime, nil
+
+			if scanTime.After(startScanTime) && scanTime.Before(endScanTime) {
+				return scanTime, local, nil
 			}
 		}
 	}
-	return time.Time{}, &domain.ScanIDForLastScanNotInAssetHistory{ScanID: scanID, AssetID: asset.ID, AssetIP: asset.IP, AssetHostname: asset.HostName}
+	return time.Time{}, unknown, &domain.ScanIDForLastScanNotInAssetHistory{ScanID: scanInfo.ScanID, AssetID: asset.ID, AssetIP: asset.IP, AssetHostname: asset.HostName}
 }
 
 // assetPayloadToAssetEvent translates a Nexpose Asset API response payload
 // into an AssetEvent for downstream services.
-func (v *NexposeAssetValidator) assetPayloadToAssetEvent(asset domain.Asset, scanTime time.Time) (domain.AssetEvent, error) {
+func (v *NexposeAssetValidator) assetPayloadToAssetEvent(asset domain.Asset, scanTime time.Time, scanType string) (domain.AssetEvent, error) {
```

Comment:
task: again, here, if `ScanType` implements `Stringer`, you can use the custom type here, which would be a good improvement, I think.

Reply:
done!